src/Microsoft.AspNet.Security/Infrastructure/AuthenticationHandler.cs

```diff
@@ -307,7 +311,18 @@ protected virtual void ApplyResponseCore()
         protected virtual async Task ApplyResponseCoreAsync()
         {
             await ApplyResponseGrantAsync();
-            await ApplyResponseChallengeAsync();
+
+            // If authenticate was called and the the status is still 401, authZ failed so set 403 and stop
+            // REVIEW: Does this need to ensure that there's the 401 is challenge for this auth type?
```

Comment:
Yeah I'm going to fix 401-403 that in a separate PR, I'll push the first set of AuthZ changes once the matching MVC PR is signed off.  Then a PR once everything is working again with the renames.  Then finally 401-403 and claims xformation will be their own hopefully smaller PRs.